command/agent/config.go

```diff
@@ -1424,6 +1468,19 @@ func (a *ServerConfig) Merge(b *ServerConfig) *ServerConfig {
 		result.DefaultSchedulerConfig = &c
 	}
 
+	if b.Search != nil {
+		result.Search = &Search{FuzzyEnabled: b.Search.FuzzyEnabled}
+		if b.Search.LimitQuery > 0 {
+			result.Search.LimitQuery = b.Search.LimitQuery
+		}
+		if b.Search.LimitResults > 0 {
```

Comment:
Should we validate a relationship between `LimitQuery` and `LimitResults`, even if just to warn in the logs that it's unexpected to have `LimitResults > LimitQuery`?

Reply:
It could actually be the case where the number of results exceeds `LimitQuery` - because of the way the `jobs` type is expanded into its subtypes (`groups`, `tasks`, `services`, `images`, `commands`, `classes`). There could be only 1 job registered, but contains more of any of those subtypes than `LimitResults`.